### ISSUE TITLE
DataAnnotations [DisplayFormat(ConvertEmptyStringToNull = false)] RTM

### ISSUE THREAD

**Issue by dpedrelli** [aspnet/Mvc#5086]
Hi,

I, recently, upgraded from RC2 to RTM. Since upgrading, [DisplayFormat(ConvertEmptyStringToNull = false)] on my models no longer works. If the user submits a view with an empty that maps to a string property, the value is set to NULL. I have to write code to test each string property for NULL and set it to an empty string.

Thanks.

**Comment by Muchiachio** [comment]
Yes, there seems to be a bug in [SimpleTypeModelBinder](https://github.com/aspnet/Mvc/blob/27a641f4bfa6e2ad65f5b9f104e6407ede28c6b1/src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/SimpleTypeModelBinder.cs#L48-L67), empty or whitespace string `value` never get's assigned to binded `model` regardless of `ConvertEmptyStringToNull`.

**Comment by dpedrelli** [comment]
Is there a workaround to get this to work, other than testing every model, before updating?

**Comment by Muchiachio** [comment]
I'm using custom model binder which is registered only for string type. You can see an example [here](https://github.com/NonFactors/MVC6.Template/blob/develop/src/MvcTemplate.Components/Mvc/Binders/TrimmingModelBinder.cs) it does trimming, but you can change the code for only binding string.

**Comment by dpedrelli** [comment]
Thanks.  That looks like a better way to handle it than data annotations.  How do I register it?

I created TrimmingModelBinder and TrimmingModelBinderProvider.  I don't test for the data annotation.  I just removed the code that returns NULL.  I tried registering TrimmingModelBinderProvider, in Startup.cs, by with services.AddMvc(options =>{options.ModelBinderProviders.Insert(0, new TrimmingModelBinderProvider());}); and services.Configure<MvcOptions>(options => { options.ModelBinderProviders.Insert(0, new TrimmingModelBinderProvider()); });  However, the binder is never called and my model contains NULLs.

Thanks.
